AK/TypeCasts.h

```diff
@@ -40,7 +40,7 @@ ALWAYS_INLINE CopyConst<InputType, OutputType>* as_if(InputType& input)
 {
     if (!is<OutputType>(input))
         return nullptr;
-    if constexpr (IsBaseOf<InputType, OutputType>) {
+    if constexpr (IsBaseOf<RemoveCVReference<InputType>, RemoveCVReference<OutputType>>) {
```

Comment:
I believe `__is_base_of` [ignores](<https://gcc.gnu.org/onlinedocs/gcc/Type-Traits.html#index-_005f_005fis_005fbase_005fof>) top-level CV qualifications?

Reply:
`RemoveReference<T>` might still be useful though?

Reply:
After going to the trouble of checking how often we were doing a `static_cast` vs `dynamic_cast`, it turns out that we could be doing a lot more static casts. My latest push replaces this with a `requires` clause that guarantees a static cast is used when possible.